shell/src/main/java/alluxio/cli/Check.java

```diff
@@ -0,0 +1,238 @@
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.cli;
+
+import alluxio.Configuration;
+import alluxio.PropertyKey;
+
+import com.google.common.base.Preconditions;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.TreeMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Utility for checking Alluxio environment.
+ */
+public final class Check {
+  private static final String USAGE = "USAGE: Check [task full name or prefix]\n\n"
+      + "Check environment for Alluxio";
+
+  private static final Options OPTIONS = new Options();
+
+  private static final Map<String, CheckTask> TASK_MAP = initializeTaskMap();
+
+  private static Map<String, CheckTask> initializeTaskMap() {
+    Map<String, CheckTask> taskMap = new TreeMap<>();
+    taskMap.put("master.port.available", new MasterRpcCheckTask());
+    taskMap.put("master.workers.ssh.reachable", new MasterWorkersSshCheckTask());
+    taskMap.put("worker.port.available", new WorkerRpcCheckTask());
+    return taskMap;
+  }
+
+  private interface CheckTask {
+    boolean check();
+  }
+
+  private static boolean isLocalPortAvailable(int port) {
+    try (ServerSocket ss = new ServerSocket(port)) {
+      ss.setReuseAddress(true);
+      return true;
+    } catch (IOException e) {
+      return false;
+    }
+  }
+
+  private static boolean isAddressReachable(String hostname, int port) {
+    try (Socket socket = new Socket(hostname, port)) {
+      return true;
+    } catch (IOException e) {
+      return false;
+    }
+  }
+
+  private static boolean isAlluxioRunning(String className) {
+    String[] command = {"bash", "-c",
+                        "ps -Aww -o command | grep -i \"[j]ava\" | grep " + className};
+    try {
+      Process p = Runtime.getRuntime().exec(command);
+      try (InputStreamReader input = new InputStreamReader(p.getInputStream())) {
+        if (input.read() >= 0) {
+          return true;
+        }
+      }
+      return false;
+    } catch (IOException e) {
+      System.err.format("Unable to check Alluxio status: %s.%n", e.getMessage());
+      return false;
+    }
+  }
+
+  private static class MasterRpcCheckTask implements CheckTask {
+
+    public MasterRpcCheckTask() {}
+
+    @Override
+    public boolean check() {
+      if (isAlluxioRunning("alluxio.master.AlluxioMaster")) {
```

Comment:
It's better to get the classname from the class, instead of hardcoding the name.